dask/array/core.py

```diff
@@ -2096,6 +2106,45 @@ def to_zarr(arr, url, component=None, storage_options=None,
     return store(arr, z, compute=compute, return_stored=return_stored)
 
 
+def _check_regular_chunks(chunkset):
+    """Check if the chunks are regular
+
+    "Regular" in this context means that along every axis, the chunks all
+    have the same size, except the last one, which may be smaller
+
+    Parameters
+    ----------
+    chunkset: tuple of tuples of ints
+        From the ``.chunks`` attribute of an ``Array``
+
+    Returns
+    -------
+    True if chunkset passes, else False
+
+    Examples
+    --------
+    >>> arr = da.zeros(10, chunks=(5, ))
+    >>> _check_regular_chunks(arr.chunks)
+    True
+
+    >>> arr = da.zeros(10, chunks=((3, 3, 3, 1), ))
+    >>> _check_regular_chunks(arr.chunks)
+    True
+
+    >>> arr = da.zeros(10, chunks=((3, 1, 3, 3), ))
+    >>> _check_regular_chunks(arr.chunks)
+    False
+    """
+    for chunks in chunkset:
+        if len(chunks) < 2:
```

Comment:
Minor point: From a readability perspective `== 1` is a bit clearer.

Reply:
Is it possible to have an empty array with no chunks?

Reply:
Good question! So if it is an empty array, `chunkset` is an empty `tuple` meaning we don't enter this `for`-loop. If it is a 0-D array, then we have `chunkset` as `((0,),)`. So `chunks` would be length `1`. For any higher dimensionality array, there would need to at least be one chunk per dimension with some length meaning each would be at least length `1`.